codegen/object.go

```diff
@@ -215,12 +215,13 @@ func (f *Field) doWriteJson(val string, remainingMods []string, astType *ast.Typ
 
 		return tpl(`{{.arr}} := graphql.Array{}
 			for {{.index}} := range {{.val}} {
-				{{.arr}} = append({{.arr}}, func() graphql.Marshaler {
-					rctx := graphql.GetResolverContext(ctx)
-					rctx.PushIndex({{.index}})
-					defer rctx.Pop()
+				{{.arr}} = append({{.arr}}, func(ctx context.Context) graphql.Marshaler {
+					rctx := &graphql.ResolverContext{
```

Comment:
this is strange. array unmarshalling isn't a resolver, and it shouldn't need a new context?

Reply:
yes, it is right.
this change required `default:` case under `case len(remainingMods) > 0 && remainingMods[0] == modList:`.
I'll move this code to right place and add some arguments to method signature.